### PR TITLE
Increase flexibility around proto keywords when parsing

### LINKED ISSUE #1797: Allow `service` for enum name
Service could not be declared inside enum block, but proto parser fails when encountering `service` as enum entry decalration.

Example:
```
enum ServiceType {
    service = 0; // parser fails, as it expects service declaration
    external = 1;
    mysql = 2;
    mdb_mysql = 3;
    conductor = 4;
    jenkins = 5;
}
```

### PR DESCRIPTION
**Pull request by oldergod**
We had a few cases where we would "teach" the consumer how to write protos, I don't think they're really valuable? so I removed them.

fix #1797

### CODE REVIEW

File: wire-library/wire-schema/src/jvmTest/kotlin/com/squareup/wire/schema/internal/parser/ProtoParserTest.kt [hunk 1]
```diff
@@ -618,21 +617,6 @@ class ProtoParserTest {
     }
   }
 
-  @Test
-  fun syntaxInWrongContextThrows() {
```
Comment (swankjesse) [inline wire-library/wire-schema/src/jvmTest/kotlin/com/squareup/wire/schema/internal/parser/ProtoParserTest.kt:622]: 👍

File: wire-library/wire-schema/src/jvmTest/kotlin/com/squareup/wire/schema/internal/parser/ProtoParserTest.kt [hunk 2]
```diff
@@ -2413,13 +2397,22 @@ class ProtoParserTest {
     assertThat(ProtoParser.parse(location, proto)).isEqualTo(expected)
   }
 
-  @Test @Ignore("Broken. See https://github.com/square/wire/issues/1797")
-  fun protoKeywordAsEnumConstants() {
+  @Test fun protoKeywordAsEnumConstants() {
+    // Note: cases were protoc fails are commented out.
```
Comment (swankjesse) [inline wire-library/wire-schema/src/jvmTest/kotlin/com/squareup/wire/schema/internal/parser/ProtoParserTest.kt:2401]: lets make Wire consistent with protoc here? Totally sucks to define an enum, use it with Wire, and then to find out your schema isn’t portable to protoc

File: wire-library/wire-schema/src/jvmTest/kotlin/com/squareup/wire/schema/internal/parser/ProtoParserTest.kt [hunk 3]
```diff
@@ -2429,16 +2422,62 @@ class ProtoParserTest {
...
+                    EnumConstantElement(location.at(13, 3), "extensions", 11),
+                )
+            )
+        )
+    )
+    assertThat(ProtoParser.parse(location, proto)).isEqualTo(expected)
+  }
+
+  @Test fun protoKeywordAsMessageNameAndField() {
+    // Note: cases were protoc fails are commented out.
```
Comment (swankjesse) [inline wire-library/wire-schema/src/jvmTest/kotlin/com/squareup/wire/schema/internal/parser/ProtoParserTest.kt:2444]: Is wire consistent with protoc?

File: wire-library/wire-schema/src/jvmTest/kotlin/com/squareup/wire/schema/internal/parser/ProtoParserTest.kt [hunk 4]
```diff
@@ -2429,16 +2422,62 @@ class ProtoParserTest {
...
+    val proto = """
+      |message message {
+      |  optional message message = 1;
```
Comment (swankjesse) [inline wire-library/wire-schema/src/jvmTest/kotlin/com/squareup/wire/schema/internal/parser/ProtoParserTest.kt:2447]: what about proto3?

**Review by swankjesse — APPROVED** [review]

File: wire-library/wire-schema/src/jvmTest/kotlin/com/squareup/wire/schema/internal/parser/ProtoParserTest.kt (hunk 2 above)
Comment (oldergod) [reply wire-library/wire-schema/src/jvmTest/kotlin/com/squareup/wire/schema/internal/parser/ProtoParserTest.kt:2401]: I see. Need to make sure then fail in both or pass in both.

File: wire-library/wire-schema/src/jvmTest/kotlin/com/squareup/wire/schema/internal/parser/ProtoParserTest.kt (hunk 4 above)
Comment (oldergod) [reply wire-library/wire-schema/src/jvmTest/kotlin/com/squareup/wire/schema/internal/parser/ProtoParserTest.kt:2447]: this is the one below commented out, it fails on both protoc/wire

File: wire-library/wire-schema/src/jvmTest/kotlin/com/squareup/wire/schema/internal/parser/ProtoParserTest.kt (hunk 2 above)
Comment (oldergod) [reply wire-library/wire-schema/src/jvmTest/kotlin/com/squareup/wire/schema/internal/parser/ProtoParserTest.kt:2401]: And yes they are both behaving the same.

**Comment by oldergod** [comment]
Added lot of tests. All is consistent with what Protoc does.

**Pull request merged**